crates/apollo-compiler/test_data/diagnostics/0089_fragment_type_condition_on_non_existent_types.txt

```diff
@@ -39,30 +39,61 @@
             file_id: FileId {
                 id: 86,
             },
-            offset: 97,
-            length: 44,
+            offset: 309,
+            length: 55,
         },
         labels: [
             Label {
                 location: DiagnosticLocation {
                     file_id: FileId {
                         id: 86,
                     },
-                    offset: 97,
-                    length: 44,
+                    offset: 309,
+                    length: 55,
                 },
-                text: "`MissingSecondType` is defined here but not declared in the schema",
+                text: "`Interface2` is defined here but not declared in the schema",
             },
         ],
         help: Some(
-            "consider defining `MissingSecondType` in the schema",
+            "consider defining `Interface2` in the schema",
         ),
         data: InvalidFragment {
             ty: Some(
-                "MissingSecondType",
+                "Interface2",
             ),
         },
     },
+    ApolloDiagnostic {
+        cache: {
+            0: "built_in_types.graphql",
+            86: "0089_fragment_type_condition_on_non_existent_types.graphql",
+        },
+        location: DiagnosticLocation {
+            file_id: FileId {
+                id: 86,
+            },
+            offset: 361,
+            length: 1,
+        },
+        labels: [
+            Label {
+                location: DiagnosticLocation {
+                    file_id: FileId {
+                        id: 86,
+                    },
+                    offset: 361,
+                    length: 1,
+                },
+                text: "`a` field is not defined",
+            },
+        ],
+        help: Some(
+            "`a` is not defined on `Interface2` type",
+        ),
+        data: UndefinedField {
+            field: "a",
+        },
+    },
```

Comment:
I feel like this diagnostic here should not be here! something went astray with fragment validation 😬

Reply:
fixed this also!